### PR TITLE
Add unit tests for ecp_mod_p192k1()

### LINKED ISSUE #7256: Add unit tests for ecp_mod_p192k1()
The task is to add unit tests for ecp_mod_p192k1(). This function reduces the input modulo secp192k1_p. The result of ecp_mod_p192k1() is not always canonical, use mpi_mod_mpi() to get a canonical representative for easier testing.

Only cases where the input is non-negative and at most twice as long as the modulus. (This is because soon these will be replaced by their fixed width _raw counterparts.)

### PR DESCRIPTION
**Pull request by gabor-mezei-arm**
## Description

The task is to add unit tests for ecp_mod_p192k1(). This function reduces the input modulo secp192k1_p.

Resolve #7256 

## Gatekeeper checklist

- [x] **changelog** not required
- [x] **backport** not required
- [x] **tests** provided

### CODE REVIEW

File: library/ecp_curves.c [hunk 1]
```diff
@@ -5594,6 +5596,12 @@ static inline int ecp_mod_koblitz(mbedtls_mpi *N, mbedtls_mpi_uint *Rp, size_t p
  * with R = 2^32 + 2^12 + 2^8 + 2^7 + 2^6 + 2^3 + 1 = 0x0100001119
  */
 static int ecp_mod_p192k1(mbedtls_mpi *N)
+{
+    return mbedtls_ecp_mod_p192k1(N);
+}
+
+MBEDTLS_STATIC_TESTABLE
+int mbedtls_ecp_mod_p192k1(mbedtls_mpi *N)
```
Comment (minosgalanakis) [inline library/ecp_curves.c:5604]: An alternative way would be to rename `ecp_mod_p192k1` -> `mbedtls_ecp_mod_p192k1` and guard it by `MBEDTLS_STATIC_TESTABLE`

But they are both functionally equivalent and those methods will soon be removed anyway, so this should not be a blocker

Comment (gabor-mezei-arm) [reply library/ecp_curves.c:5604]: I think it is easier to introduce a new function and when updating to `_raw` function only a rename must be done.

**Comment by gabor-mezei-arm** [comment]
Rebased

File: tests/suites/test_suite_ecp.function [hunk 2]
```diff
@@ -1481,6 +1481,48 @@ exit:
...
+    size_t limbs = limbs_N;
+    size_t bytes = limbs * sizeof(mbedtls_mpi_uint);
+
+    TEST_EQUAL(limbs_X, 2 * limbs);
+    TEST_EQUAL(limbs_res, limbs);
+
+    TEST_EQUAL(mbedtls_ecp_mod_p192k1(&X), 0);
+    TEST_EQUAL(mbedtls_mpi_mod_mpi(&X, &X, &N), 0);
+    TEST_LE_U(mbedtls_mpi_core_bitlen(X.p, limbs_X), 192);
+    ASSERT_COMPARE(X.p, bytes, res.p, bytes);
```
[the remark this answers is not in the document]
Comment (gabor-mezei-arm) [reply tests/suites/test_suite_ecp.function:1388]: The `mbedtls_mpi_mod_mpi()` is guaranteeing the canonical form.

**Review by minosgalanakis — APPROVED** [review]
Looks good

**Review by minosgalanakis — APPROVED** [review]
LGTM

**Comment by yanesca** [comment]
Just a reminder: all PRs related to this quarter's bignum work should have the `priority-high` label and added to the unified board.

**Review by yanesca — APPROVED** [review]
LGTM

**Pull request merged**